xarray/tests/test_groupby.py

```diff
@@ -202,4 +202,21 @@ def test_da_groupby_assign_coords():
     assert_identical(expected, actual2)
 
 
+def test_da_groupby_cumsum():
+    ds = xr.Dataset(
+        {"foo": (("x",), [7, 3, 1, 1, 1, 1, 1])},
+        coords={"x": [0, 1, 2, 3, 4, 5, 6], "group_id": ("x", [0, 0, 1, 1, 2, 2, 2])},
+    )
+    actual = ds.groupby("group_id").cumsum(dim="x")
+    expected = xr.Dataset(
+        {
+            "foo": (("x",), [7, 10, 1, 2, 1, 2, 3]),
+            "group_id": (("x",), [0, 0, 1, 1, 2, 2, 2]),
```

Comment:
Any idea why `group_id` becomes a data variable rather than a coord?

Here's the equivalent with `sum`:
```python
In [6]: ds.groupby("group_id").sum()
Out[6]:
<xarray.Dataset>
Dimensions:   (group_id: 3)
Coordinates:
  * group_id  (group_id) int64 0 1 2
Data variables:
    foo       (group_id) int64 10 2 3
```

Reply:
`group_id` is removed from coords in this line https://github.com/pydata/xarray/blob/6b70107ab3063187b663290538c0d5a4107dab6e/xarray/core/merge.py#L363
If we comment this line and run our example again, group_id appears in coords after cumsum. I  need some help to understand what this line does.

Reply:
I could look more. 

Does anyone who have more experience with this code know what's happening?